<?xml version="1.0"?>
<launch>

    <node pkg="love_maker_2000" type="cylinder_detector" name="cylinder_detector">

        <remap from="input" to="/love_maker_2000/voxelgrid"/>

        <param name="normal_distance_weight" value="0.1" />  <!-- 0.1 -->
        <param name="max_iterations" value="10000" />   <!-- 10000 -->
        <param name="distance_threshold" value="0.05" />   <!-- 0.05 -->
        <param name="radius_min" value="0.11" />    <!-- 0.06 -->
        <param name="radius_max" value="0.13" />    <!-- 0.2 -->
        <param name="cylinder_points_threshold" value="700"/>

        <param name="plane_points_threshold" value="500" />

        <param name="torus_ransac_max_iterations" value="1000" />
    </node>

    <!-- Node that makes the cloud point sparser -->
    <node pkg="love_maker_2000" type="voxelgrid_new" name="voxelgrid" output="screen">

        <remap from="input" to="/camera/depth/points"/>
        <remap from="output" to="/love_maker_2000/voxelgrid"/>

        <!-- Leaf size in meters -->
        <param name="leaf_size_x" value="0.01"/> <!-- 0.01 -->
        <param name="leaf_size_y" value="0.01"/> <!-- 0.01 -->
        <param name="leaf_size_z" value="0.01"/> <!-- 0.01 -->

    </node>

</launch>
